### ISSUE TITLE
Can We Not Have Diving And Descending On The Same Button, Please?

### ISSUE THREAD

**Issue by EternaDash** [Sollace/Unicopia#347]
So Basically, I'm Flying Into A Cave System To Go Mining. I Hold Crouch To Descend Slowly Into It, Very Simple. Except I Dive Down At The Speed Of Sound, And I Slam My Face Into Stone So Fast I Become A Red Mist.

I Don't Know How The Diving Thing Works, But I Feel Like It Should Have It's Own Button Rather Than Working With Crouch, Since I Would Prefer My Health Not Get Obliterated For Just Trying To Go Down. Could Also Be A Muscle Memory Thing Since I'm Used To 1.3.3, But Being Able To Choose Between Going Down Faster And Zooming Into The Ground At Lightspeed Would Be Nice.
_i feel bad for anyone in hardcore that this happens to, oh that has to sting._

And Just To Make Sure: No, There Was No Weather In Effect, The Sky Was Perfectly Clear.

**Comment by PrinceDrowsy** [comment]
It might be better to have diving take a little bit longer to kick in stead, and increase in speed more gradually, since it's intended to be a result of prolonged descending without catching yourself by flapping your wings. As is in the current build, you can nullify the diving by remembering to periodically flap to slow your descend, which I enjoy, however I do agree that it could be a bit more gradual the transitioning into an increase in speed to make keeping track of this easier, and prevent accidents in tighter spaces.

**Comment by PrinceDrowsy** [comment]
Actually, it might be better to instead have diving activate by ceasing flight entirely and falling from a significant height, since pegasi in the show and most birds in real life close their wings against their bodies completely in order to dive. Upon ceasing flight and commencing free-fall, a pegasus pony's falling speed will begin to increase over the course of the dive until they build up enough to perform the trick in question (such as a Sonic Rainboom), or they can simply open their wings again without the additional fanfare and use it as a temporary natural speed boost to their flight.

**Comment by Sollace** [comment]
Sneaking whilst in the air causes the player to move down. This is part of the base game's code that it uses for creative flight that I can't override. I had previously tried to counter it by adding an upwards velocity when the player was sneaking, but that resulted in the random bumpiness that I struggled to debug for a long time.

When I fixed the bumpiness I decided not to fight it and make it an intended feature.

Descending happens naturally if you're moving forward and don't flap your wings to keep going up (as is expected for self-powered flight).
Diving happens when you sneak to push yourself downwards. From my testing I do not see anything wrong with the rate diving moves you down. It's perfectly controllable and works well with the low-flying mechanic.

Considering there is a massive wall of text [here](https://github.com/Sollace/Unicopia/issues/356) asking about how pegasi can be nerfed, I'm not very inclined to turn around and buff them. 

tl;dr If you don't want to go down, don't hit the button that makes you go down.
